src/agent/errors.ts

```diff
@@ -2,30 +2,9 @@ import { ErrorPolykey, sysexits } from '../errors';
 
 class ErrorAgent<T> extends ErrorPolykey<T> {}
 
-class ErrorAgentRunning<T> extends ErrorPolykey<T> {
-  static description = 'Agent Client is running';
-  exitCode = sysexits.USAGE;
-}
-
-class ErrorAgentClientNotStarted<T> extends ErrorAgent<T> {
-  static description = 'Agent Client is not started';
-  exitCode = sysexits.USAGE;
-}
-
-class ErrorAgentClientDestroyed<T> extends ErrorAgent<T> {
-  static description = 'Agent Client is destroyed';
-  exitCode = sysexits.USAGE;
-}
-
 class ErrorConnectionInfoMissing<T> extends ErrorAgent<T> {
-  static description = 'Vault already exists';
+  static description = 'Connection info was missing from connection metadata';
   exitCode = sysexits.UNAVAILABLE;
 }
 
-export {
-  ErrorAgent,
-  ErrorAgentClientNotStarted,
-  ErrorAgentRunning,
-  ErrorAgentClientDestroyed,
```

Comment:
If we no longer have these exceptions, does that mean there is no "client" instance for RPC with respect to the client service or agent service?

I thought we would still have a RPC client instance somewhere? Does that not require to be a `StartStop` situation?

Reply:
I'm not sure what these exceptions were for, So far as I can tell they weren't used for a long time. In fact, I'm not even sure this file is needed anymore. The only exception left here isn't used.

Reply:
The RPC has it's own lifecycle errors, same for js-quic or websocket.

Reply:
Should RPC client's own lifecycle errors be specialised for both client and agent?

Reply:
There are no `rpcClientClient` anymore. That's handled by a `RPCClient` or `RpCServer` with a client and agent maniefest. There's no need for a derivative class anymore.

As for `ErrorConnectionInfoMissing`, it depends on the context. It could be missing in the nodes domain when verifying or making use of the connection information. But it could also be missing when needed in the agent handlers.

During normal usage this should never be missing since it is checked at multiple levels. In order they are the Quic system, Nodes domain and then agent handlers. In reality the Quic system should ensure that this information exists and the only part that would be missing is the remote certificates and NodeId. 

For the nodes and handlers the error is only really used as a fail safe catch and enforcing the type. It could be handled by a `never` error but specific errors here give better context. So I'd rather keep them separate.

Reply:
I'm going to remove the nodes domain error. anywhere it could be used would be a logic failure. I'll use never in these cases. If the remote certs are missing then it should be caught by the custom verification logic. Any active connection or stream SHOULD have this information available.

I'll keep the error for the agent handlers since it's possible for bad usage or through tests for the remote information to be missing in that context.

Reply:
> During normal usage this should never be missing since it is checked at multiple levels. In order they are the Quic system, Nodes domain and then agent handlers. In reality the Quic system should ensure that this information exists and the only part that would be missing is the remote certificates and NodeId.

If the QUIC system checks for this information, then why do we need this error then? We can just assume in PK that this info always exists. Then just go to `js-quic` and ensure that the info is provided.

Reply:
Fixed.